src/main.cpp

```diff
@@ -2701,11 +2697,10 @@ bool AcceptBlock(CBlock& block, CValidationState& state, CBlockIndex** ppindex,
     return true;
 }
 
-static bool IsSuperMajority(int minVersion, const CBlockIndex* pstart, unsigned int nRequired)
+static bool IsSuperMajority(int minVersion, const CBlockIndex* pstart, unsigned nRequired, const Consensus::Params& consensusParams)
```

Comment:
As said, at the beginning it was getting nRequired and nToCheck, but @theuni suggested the change. I'm happy to change it back but I don't have strong feelings either way.
You could also have a version with nRequired and nToCheck and two other inlined functions:

```
bool IsSuperMajorityUpgrade(int minVersion, const CBlockIndex* pstart, const Consensus::Params& consensusParams);
bool IsSuperMajorityRejectOld(int minVersion, const CBlockIndex* pstart, const Consensus::Params& consensusParams); 
```
that call the Consensus::Params-independent IsSuperMajority.

That should also make things easier for replacing the softfork feature voting since one would only need to touch inside those methods and Consensus::Params.

I think those are all options but I'm happy just decoupling IsSuperMajority from the Params() factory, so if anyone has stronger opinions between the 3 options I don't really care which one to use.